### ISSUE TITLE
Investigate building igc connector image based on maven artifacts or multi-stage builds

### ISSUE THREAD

**Issue by planetf1** [odpi/egeria#638]
Note: I have now changed the way the IGC connector/adapter is used so that we work off a full egeria code image.

However when I originally coded the container I decided to pull the relevant igc artifacts directly. Maven should already handle all required dependencies (and this can be seen when the commands are run)

```
mvn -DremoteRepositories=https://nexus.odpi.org/content/repositories/snapshots/ org.apache.maven.plugins:maven-dependency-plugin:3.1.1:get -Dartifact=org.odpi.egeria:igc-repository-proxy:0.3-SNAPSHOT
mvn install org.apache.maven.plugins:maven-dependency-plugin:3.1.1:copy -Dartifact=org.odpi.egeria:igc-repository-proxy:0.3-SNAPSHOT -DoutputDirectory=.
java -jar ./igc-repository-proxy-0.3-SNAPSHOT.jar
```

However when run, the spring boot application starts, and some URLs are available, however some REST apis are not available -- for example

```
POST /open-metadata/admin-services/users/egeria/servers/myserver/local-repository/mode/ibm-igc/details HTTP/1.1
Accept: application/json, */*
Accept-Encoding: gzip, deflate
Authorization: Basic YWRtaW46YWRtaW4=
Connection: keep-alive
Content-Length: 93
Content-Type: application/json
Host: localhost:9080
User-Agent: HTTPie/1.0.2

{
    "igcAuthorization": "aXNhZG1pbjppc2FkbWluCg==",
    "igcBaseURL": "https://158.176.66.119:9446"
}

HTTP/1.1 404
Content-Type: application/json;charset=UTF-8
Date: Thu, 24 Jan 2019 20:38:46 GMT
Transfer-Encoding: chunked

{
    "error": "Not Found",
    "message": "No message available",
    "path": "/open-metadata/admin-services/users/egeria/servers/myserver/local-repository/mode/ibm-igc/details",
    "status": 404,
    "timestamp": "2019-01-24T20:38:47.059+0000"
}
```
 It's possible a silly mistake was made - the jar appears to contain the connector and adapter. Maybe the spring component scan is failing.

It would be useful to get our assemblies & maven artifacts clarified to allow this kind of useage. It could for example help when building docker images, as one could take the BUILT artifacts as needed (different containers will need differing subsets) and quickly create an image - or even retrieve at runtime, as opposed to doing a full build.

Opening for investigation, but alternative approach taken for now.

**Comment by mandy-chessell** [comment]
The IGC Repository Proxy is a different server from the server chassis and has its own implementation of the admin services.  This implementation does not include this REST Operation
/open-metadata/admin-services/users/egeria/servers/myserver/local-repository/mode/ibm-igc/details.

The IGC Repository Server Proxy is now not needed.  I will take it out of the assemblies and once you are happy with Chris's new implementation of the IGC repository connector/event mapper I will delete the igc-repository-proxy artifact from the Egeria build.

For interest, here is the implementation of the igc-repository-proxy admin services:
https://github.com/odpi/egeria/blob/master/open-metadata-resources/open-metadata-demos/virtual-data-connector/igc-repository-proxy/src/main/java/org/odpi/openmetadata/virtualdataconnector/igc/admin/IGCRepositoryProxyAdminResource.java

**Comment by planetf1** [comment]
The current helm charts use the new implementation running within the server chassis. The code is built within the container from source.

This is done 'by hand' though I am also looking at redhat's source-to-image as part of the 'pipeline'

 I'll also again evaluate instead using maven artifacts, as this reduces image build time & size - but the penalty is less currency. I suggest leaving this issue open to support this evaluation

**Comment by cmgrote** [comment]
(Renamed issue to cover what we're now using it to track -- old IGC connector is now removed, I think? so leaving open to a race between @planetf1 and @cmgrote as to who gets to this optimisation of the images first 😜)

**Comment by planetf1** [comment]
I investigated the approach of using maven artifacts for ranger (and posted to the ranger-dev mailing list). See https://lists.apache.org/thread.html/9aeaac8c04430a9143985ae9a958321d5f7546c481f7ace1c6c7fd84@%3Cdev.ranger.apache.org%3E

In Ranger's case what I found was that whilst I could script retrieval of the artifacts using an improvement of what I originally tried for egeria/igc, the artifacts I downloaded were incomplete. They support building, and runtime libraries.. but don't necessarily include everything like scripts.

So for ranger I opted to stop -- and stick with building from source. If any responses come on the mailing list I'll re-evaluate. It would also be interesting to see if anyone else has tried/is interested in this report. We *could* decide ourselves to do this for egeria, but perhaps for now it's easier to stick with using the 'assemblies' that are built, and to ensure they are complete.

Optimizing the image builds using multi stage builds is viable.. but is probably covered elsewhere, so it may make sense to close off this issue? 

I am making image changes currently, but to aid in verification and debugging rather than build optimization (so as to complete the set of components in vdc)
